biz.aQute.bndlib/src/aQute/bnd/osgi/PluginsContainer.java

```diff
@@ -356,14 +360,15 @@ private Object loadPlugin(Processor processor, ClassLoader loader, Attrs attrs,
 		boolean ignoreError) {
 		try {
 			Class<?> c = loader.loadClass(className);
-			Object plugin = publicLookup().findConstructor(c, defaultConstructor)
-				.invoke();
-			processor.customize(plugin, attrs, this);
-			add(plugin);
-			if (plugin instanceof AutoCloseable) {
-				closeablePlugins.add((AutoCloseable) plugin);
+			if (c.isInterface()) {
+				interfaces.put(c, attrs);
```

Comment:
If the type is an interface, this method returns `null`. This causes `loadPlugins` to try again using the pluginpath which will also return `null` since the interface is still an interface. If the plugin declaration has a `command` directive, then it will end up in the missing commands set. Wouldn't it be better to avoid the retry?